crates/primitives/src/transaction/mod.rs

```diff
@@ -899,6 +899,16 @@ impl Encodable for TransactionSignedEcRecovered {
     }
 }
 
+impl Decodable for TransactionSignedEcRecovered {
+    fn decode(buf: &mut &[u8]) -> Result<Self, DecodeError> {
+        let signed_transaction = TransactionSigned::decode(buf)?;
+        let signer = signed_transaction
+            .recover_signer()
+            .ok_or(DecodeError::Custom("Unable to recover decoded transaction signer."))?;
```

Comment:
I guess this is fine, since this is basically just a convenience impl

wdyt @Rjected

Reply:
yeah, this seems alright as a convenience impl